win32ss/gdi/ntgdi/palette.c

```diff
@@ -769,13 +769,8 @@ IntGdiRealizePalette(HDC hDC)
 
     ASSERT(ppalDC->flFlags & PAL_INDEXED);
 
-    // FIXME: Should we resize ppalSurf if it's too small?
-    realize = (ppalDC->NumColors < ppalSurf->NumColors) ? ppalDC->NumColors : ppalSurf->NumColors;
-
-    for (i=0; i<realize; i++)
-    {
-        InterlockedExchange((LONG*)&ppalSurf->IndexedColors[i], *(LONG*)&ppalDC->IndexedColors[i]);
-    }
```

Comment:
Does it change something if you disable this code only for memory DCs ?

Reply:
> I downloaded the full output from the testbot runs for both VBox and KVM and did a search for "RealizePalette unimplemented" which are the first two words in both of the DPRINT1's that were being used at that time to show memory DC's versus device DC's and there were zero (0) instances found. So this shows me that our regression tests do never exercise this code path. We have three test cases of which I am aware. These are 1) Durak, 2) GDIProg-Menu, and 3) GDIProg-Palette that use this code and these all use "memory DC's".

So I believe that disabling this code only for memory DC's would fix all known instances of problems with this code for now.

It would be good to have regression tests that exercise RealizePalette for both of these types of DC's, but I have no idea how to create them for now. Thanks.

Reply:
@zefklop, I did a search for RealizePalette in the regression tests and it look like you had started to create tests for this in 'modules\rostests\apitests\win32nt\ntuser\NtUserSelectPalette.c', but this code is #if 0'ed out. Maybe you have a better idea of how to finish these tests now? Thanks.

Reply:
I cleanup up this regression test code enough to get it to compile and when run it shows:
```
WARNING:  IntGdiRealizePalette at win32ss/gdi/ntgdi/palette.c:755 is UNIMPLEMENTED!
```
This is triggered when 'pdc->dctype == DCTYPE_DIRECT'
So I am not sure that we could ever reach the RealizePalette code with a "device DC".